volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -192,108 +341,90 @@ def get_requirements(cls):
             ),
         ]
 
+    @staticmethod
+    def parse_ads_data_records(
+        mft_record: interfaces.objects.ObjectInterface,
+        attr: interfaces.objects.ObjectInterface,
+        symbol_table,
+    ):
+        return MFTScan.parse_data_records(mft_record, attr, symbol_table, False)
+
     def _generator(self):
-        layer = self.context.layers[self.config["primary"]]
+        for (
+            offset,
+            rec_type,
+            rec_num,
+            attr_type,
+            file_name,
+            ads_name,
+            content,
+        ) in MFTScan.enumerate_mft_records(
+            self.context, self.config, self.config_path, self.parse_ads_data_records
+        ):
+            yield (
+                0,
+                (offset, rec_type, rec_num, attr_type, file_name, ads_name, content),
```

Comment:
This looks the same as the `for` input, wouldn't this be easier to write as:

```
for item in MFTScan.enumerate_mft_records(<args>):
  yield (0, item)
```

or have I missed something?

Reply:
Oh I see, this matches later calls which don't use all the return values, ok, that's fine.